dune/iga/nurbstrimboundary.hh

```diff
@@ -0,0 +1,139 @@
+//
+// Created by Henri on 14.03.2023.
+//
+#pragma once
+
+#include "ibraGeometry.hh"
+#include "nurbstrimutils.hh"
+
+#include <clipper2/clipper.core.h>
+#include <dune/iga/nurbspatchgeometry.h>
+
+namespace Dune::IGA {
+  
+  class Boundary {
+   public:
+    static constexpr int worldDim = 2;
+    static constexpr int gridDim = 1;
+
+    using Geometry = NURBSPatchGeometry<gridDim, worldDim>;
+    using ControlPoint = Geometry::ControlPointType;
+    using ControlPointNetType = Dune::IGA::MultiDimensionNet<gridDim, ControlPoint>;
+    using PatchData = Dune::IGA::NURBSPatchData<gridDim, worldDim>;
+
+    using Point       = Dune::FieldVector<double, worldDim>;
+    using PointVector = std::vector<Point>;
+    using Domain      = std::array<double, 2>;
+
+
+    // Member variables
+    Geometry nurbsGeometry;
+
+    Domain domain{};
+
+    // Control Points are the start and endpoint (Rename???)
+    PointVector controlPoints;
```

Comment:
You don't have to introduce an type alias everywhere. It makes the code hard to read, since you always have to look what type this actually is. thus maybe, staying with `std::vector<Point>`, is ok.